packages/vm/src/evm/opcodes/codes.ts

```diff
@@ -57,7 +57,7 @@ const opcodes: OpcodeEntry = {
   0x07: { name: 'SMOD', isAsync: false, dynamicGas: false },
   0x08: { name: 'ADDMOD', isAsync: false, dynamicGas: false },
   0x09: { name: 'MULMOD', isAsync: false, dynamicGas: false },
-  0x0a: { name: 'EXP', isAsync: false, dynamicGas: false },
+  0x0a: { name: 'EXP', isAsync: false, dynamicGas: true },
```

Comment:
I would strongly suggest to not do *any* kind of such non-related fixes along such type of work (or is there a relationship I am not seeing?). Rather open an issue and do separately later. These kind of things DO increase the complexity (there wasn't anything explicitly "not working" before) and even more increases the risk of drawing in new errors, test failures,... on top of already complex work.

Reply:
Sure, I can revoke this, then I will add a tracking issue for these inconsistencies. This is definitely a bug in the `step` event since the reported gas used there is wrong.

Reply:
Ah, not sure if a revoke is needed at this point now, maybe better decide for yourself what you think is better/more practical here for now. Just a general comment (I guess along these general refactoring one can find *a lot* of things 😋).

Reply:
I think you have raised a good point, I'm trying to cramp too much into one PR, it is cleaner for these semi-unrelated things to put into another PR :smile: